### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by Ansku**
- removed unused private methods
- removed unused private variables
- removed unnecessary initializations
- removed unnecessary substring(0)
- removed inner assignments
- renamed private methods that started with upper case
- renamed static final variables to use upper case
- converted to use non-deprecated options
- suppressed unavoidable warnings
- divided long Strings to multiple lines
- added missing types
- added missing JavaDoc parameters
- formatting
- updated comments & JavaDocs

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/framework/12333)

### CODE REVIEW

File: compatibility-server/src/main/java/com/vaadin/v7/ui/Tree.java [hunk 1]
```diff
@@ -670,8 +675,6 @@ public void paintContent(PaintTarget target) throws PaintException {
         // rendered selectedKeys
         LinkedList<String> selectedKeys = new LinkedList<String>();
```
[the remark this answers is not in the document]
Comment (Ansku) [reply compatibility-server/src/main/java/com/vaadin/v7/ui/Tree.java:676]: @Lift-dev ignore

File: compatibility-server/src/main/java/com/vaadin/v7/ui/Tree.java [hunk 2]
```diff
@@ -670,8 +675,6 @@ public void paintContent(PaintTarget target) throws PaintException {
...
 
-        final LinkedList<String> expandedKeys = new LinkedList<String>();
-
         // Iterates through hierarchical tree using a stack of iterators
         final Stack<Iterator<?>> iteratorStack = new Stack<Iterator<?>>();
```
[the remark this answers is not in the document]
Comment (Ansku) [reply compatibility-server/src/main/java/com/vaadin/v7/ui/Tree.java:679]: @Lift-dev ignore

[the remark this answers is not in the document]
Comment (Ansku) [reply compatibility-server/src/main/java/com/vaadin/v7/ui/Tree.java:679]: @muse-dev ignore

**Review by roastedcpu — APPROVED** [review]

**Review by OlliTietavainenVaadin — APPROVED** [review]
:lgtm:

> Reviewed 51 of 52 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/12333#-:-Me0-lY604Ec7Y9RputN:bnfp4nl)* status: all files reviewed, 4 unresolved discussions (waiting on @Ansku, @Muse-Dev, and @sonatype-lift[bot])

**Review by Ansku — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/12333)* status: all files reviewed, 4 unresolved discussions (waiting on @Muse-Dev and @sonatype-lift[bot])

___
*[compatibility-server/src/main/java/com/vaadin/v7/ui/Tree.java, line 679 at r2](https://reviewable.io/reviews/vaadin/framework/12333#-MdbzWR80tCTdu3OqWcl-r2-679:-Me00urxDL20LKT-6drc:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/9a2400286db2411508465ecb42497a3092df9bb8/compatibility-server/src/main/java/com/vaadin/v7/ui/Tree.java#L679)):*
<details><summary><i>Previously, sonatype-lift[bot] wrote…</i></summary><blockquote>

I've recorded this as ignored for this pull request. If you change your mind, just comment `@sonatype-lift unignore`.
</blockquote></details>

Done.
___
*[server/src/main/java/com/vaadin/server/VaadinServlet.java, line 1733 at r1](https://reviewable.io/reviews/vaadin/framework/12333#-MdbzWR80tCTdu3OqWcr-r1-1733:-Me00yf101asDHMS5PN5:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/2771adbc5b93323be57dd090d907c34851f777f4/server/src/main/java/com/vaadin/server/VaadinServlet.java#L1733)):*
<details><summary><i>Previously, sonatype-lift[bot] wrote…</i></summary><blockquote>

*OperatorPrecedence:*  Use grouping parenthesis to make the operator precedence explicit
(at-me [in a reply](https://help.sonatype.com/lift) with `help` or `ignore`)
</blockquote></details>

Done.
___
*[uitest/src/main/java/com/vaadin/tests/components/HasValueRequiredIndicator.java, line 72 at r1](https://reviewable.io/reviews/vaadin/framework/12333#-MdbzWR9GWyv_N_pLOor-r1-72:-Me0100e7Iv7QlffiE__:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/2771adbc5b93323be57dd090d907c34851f777f4/uitest/src/main/java/com/vaadin/tests/components/HasValueRequiredIndicator.java#L72)):*
<details><summary><i>Previously, sonatype-lift[bot] wrote…</i></summary><blockquote>

*ClassNewInstance:*  Class.newInstance() bypasses exception checking; prefer getDeclaredConstructor().newInstance()
(at-me [in a reply](https://help.sonatype.com/lift) with `help` or `ignore`)
</blockquote></details>

Done.

**Review by ZheSun88 — APPROVED** [review]
> Reviewed 51 of 52 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/12333)* status: all files reviewed, 4 unresolved discussions (waiting on @Muse-Dev and @sonatype-lift[bot])

**Review by ZheSun88 — APPROVED** [review]
> Dismissed @sonatype-lift[bot] from 4 discussions.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/12333)* status: all files reviewed, 1 unresolved discussion (waiting on @Ansku)

**Review by ZheSun88 — APPROVED** [review]
> Dismissed @Ansku from a discussion.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/12333)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @Ansku)

**Pull request merged**